.github/workflows/test.yaml

```
@@ -290,7 +290,30 @@ jobs:
         name: ${{ matrix.python-version || env.DEFAULT_PYTHON_VERSION }}-${{matrix.job-name || 'default-tests' }}
         fail_ci_if_error: true
 
+  benchmarks:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '${{ env.DEFAULT_PYTHON_VERSION }}'
 
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip tox~=3.0
+        tox -e benchmarks-ci
+
+    - name: Run benchmarks
+      uses: CodSpeedHQ/action@v2
+      with:
+        token: ${{ secrets.CODSPEED_TOKEN }}
+        # codspeed runs this command under a CPU emulator, so it's super-slow.
+        # As such, runnig the full `tox -e benchmarks-ci` here is too slow, so
+        # we break it into two parts.
+        run: .tox/benchmarks-ci/bin/pytest --codspeed benchmarks/
```

And then without tox we would have 

```suggestion
        run: python -m pytest --codspeed benchmarks/
```